Upgrade to 51.1 failed due to a migration error

### Describe the bug

Upgrading from 0.49.17 to 0.51.1 failed with the following error

2024-11-04 18:42:12,286 INFO liquibase :: Parsed changelog file 'liquibase.yaml'
2024-11-04 18:42:12,611 INFO db.setup :: Running Database Migrations...
2024-11-04 18:42:12,613 INFO db.setup :: Setting up Liquibase...
2024-11-04 18:42:12,948 INFO liquibase :: Parsed changelog file 'liquibase.yaml'
2024-11-04 18:42:12,982 INFO db.liquibase :: Updating liquibase table to reflect consolidated changeset filenames
2024-11-04 18:42:13,008 INFO liquibase.lockservice :: Successfully acquired change log lock
2024-11-04 18:42:13,010 INFO db.liquibase :: No migration lock found.
2024-11-04 18:42:13,011 INFO db.liquibase :: Migration lock acquired.
2024-11-04 18:42:13,041 INFO liquibase.lockservice :: Successfully released change log lock
2024-11-04 18:42:13,073 ERROR metabase.core :: Metabase Initialization FAILED
java.sql.BatchUpdateException: (conn=53968) Duplicate entry 'v00.00-000-qnkhuat-migrations/001_update_migrations.yaml' for key 'idx_databasechangelog_id_author_filename'
	at org.mariadb.jdbc.MariaDbStatement.executeBatchExceptionEpilogue(MariaDbStatement.java:323)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeBatch(ClientSidePreparedStatement.java:299)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeBatch(NewProxyPreparedStatement.java:2544)
	at clojure.java.jdbc$execute_batch.invokeStatic(jdbc.clj:598)


### To Reproduce

Upgrading from 0.49.17 to 0.51.1 


### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

Database: MariaDB

### Severity

P1

### Additional context

_No response_

Version: 0.49.25

2024-11-04 21:20:28,178 INFO db.liquibase :: Ejecutando 13 migraciones ...
2024-11-04 21:20:28,674 ERROR liquibase.changelog :: ChangeSet migrations/001_update_migrations.yaml::v49.2024-06-27T00:00:04::calherries encountered an exception.

UPDATE SUMMARY
Run:                         13
Previously run:             320
Filtered out:                 5
-------------------------------
Total change sets:          338


FILTERED CHANGE SETS SUMMARY
DBMS mismatch:                5

2024-11-04 21:20:28,772 ERROR metabase.core :: La inicialización de Metabase ha FALLADO
liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset migrations/001_update_migrations.yaml::v49.2024-06-27T00:00:04::calherries:
     Reason: liquibase.exception.DatabaseException: (conn=56584) Can't create table `metabase`.`metabase_field` (errno: 121 "Duplicate key on write or update") [Failed SQL: (1005) ALTER TABLE `metabase`.`metabase_field` ADD CONSTRAINT `fk_field_parent_ref_field_id` FOREIGN KEY (`parent_id`) REFERENCES `metabase`.`metabase_field` (`id`) ON DELETE RESTRICT]
	at liquibase.command.CommandScope.execute(CommandScope.java:253)
	at liquibase.Liquibase.lambda$update$0(Liquibase.java:245)
	at liquibase.Scope.lambda$child$0(Scope.java:186)
	at liquibase.Scope.child(Scope.java:195)
	at liquibase.Scope.child(Scope.java:185)
	at liquibase.Scope.child(Scope.java:164)
	at liquibase.Liquibase.runInScope(Liquibase.java:1419)
	at liquibase.Liquibase.update(Liquibase.java:234)
	at liquibase.Liquibase.update(Liquibase.java:212)
	at liquibase.Liquibase.update(Liquibase.java:194)
	at metabase.db.liquibase$migrate_up_if_needed_BANG_.invokeStatic(liquibase.clj:305)
	at metabase.db.liquibase$migrate_up_if_needed_BANG_.invoke(liquibase.clj:287)
	at metabase.db.setup$migrate_BANG_$fn__51646.invoke(setup.clj:80)
	at metabase.db.liquibase$do_with_liquibase$f_STAR___49147.invoke(liquibase.clj:139)
	at metabase.db.liquibase$do_with_liquibase.invokeStatic(liquibase.clj:142)
	at metabase.db.liquibase$do_with_liquibase.invoke(liquibase.clj:130)
	at metabase.db.setup$migrate_BANG_.invokeStatic(setup.clj:75)
	at metabase.db.setup$migrate_BANG_.doInvoke(setup.clj:56)
	at clojure.lang.RestFn.invoke(RestFn.java:445)
	at metabase.db.setup$run_schema_migrations_BANG_.invokeStatic(setup.clj:147)
	at metabase.db.setup$run_schema_migrations_BANG_.invoke(setup.clj:141)
	at metabase.db.setup$setup_db_BANG_$fn__51674$fn__51675.invoke(setup.clj:165)
	at metabase.util.jvm$do_with_us_locale.invokeStatic(jvm.clj:239)
	at metabase.util.jvm$do_with_us_locale.invoke(jvm.clj:225)
	at metabase.db.setup$setup_db_BANG_$fn__51674.invoke(setup.clj:160)
	at metabase.db.setup$setup_db_BANG_.invokeStatic(setup.clj:159)
	at metabase.db.setup$setup_db_BANG_.invoke(setup.clj:153)
	at metabase.db$setup_db_BANG_$fn__51694.invoke(db.clj:69)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:64)
	at metabase.db$setup_db_BANG_.invoke(db.clj:55)
	at metabase.core$init_BANG__STAR_.invokeStatic(core.clj:116)
	at metabase.core$init_BANG__STAR_.invoke(core.clj:101)
	at metabase.core$init_BANG_.invokeStatic(core.clj:159)
	at metabase.core$init_BANG_.invoke(core.clj:154)
	at metabase.core$start_normally.invokeStatic(core.clj:171)
	at metabase.core$start_normally.invoke(core.clj:165)
	at metabase.core$entrypoint.invokeStatic(core.clj:204)
	at metabase.core$entrypoint.doInvoke(core.clj:198)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
	at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.bootstrap.main(Unknown Source)
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset migrations/001_update_migrations.yaml::v49.2024-06-27T00:00:04::calherries:
     Reason: liquibase.exception.DatabaseException: (conn=56584) Can't create table `metabase`.`metabase_field` (errno: 121 "Duplicate key on write or update") [Failed SQL: (1005) ALTER TABLE `metabase`.`metabase_field` ADD CONSTRAINT `fk_field_parent_ref_field_id` FOREIGN KEY (`parent_id`) REFERENCES `metabase`.`metabase_field` (`id`) ON DELETE RESTRICT]
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:151)
	at liquibase.command.core.AbstractUpdateCommandStep.lambda$run$0(AbstractUpdateCommandStep.java:110)
	at liquibase.Scope.lambda$child$0(Scope.java:186)
	at liquibase.Scope.child(Scope.java:195)
	at liquibase.Scope.child(Scope.java:185)
	at liquibase.Scope.child(Scope.java:164)
	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:108)
	at liquibase.command.core.UpdateCommandStep.run(UpdateCommandStep.java:105)
	at liquibase.command.CommandScope.execute(CommandScope.java:217)

--- a/resources/migrations/001_update_migrations.yaml
+++ b/resources/migrations/001_update_migrations.yaml
@@ -8313,6 +8313,7 @@ databaseChangeLog:
       id: v50.2024-05-29T18:42:14
       author: johnswanson
       comment: Populate `archived_directly` and `archive_operation_id` (MySQL)
+      validCheckSum: ANY
       changes:
         - sqlFile:
             dbms: mysql

--- a/resources/migrations/trash/mysql.sql
+++ b/resources/migrations/trash/mysql.sql
@@ -87,8 +87,8 @@ WITH RECURSIVE Ancestors (id, archived, archived_directly, archive_operation_id,
     collection.archived_directly,
     parent.archive_operation_id,
     collection.location
-    FROM collection
-    JOIN Ancestors parent ON collection.location = concat(parent.location, parent.id, '/')
+    FROM Ancestors parent
+    JOIN collection ON collection.location = concat(parent.location, parent.id, '/')
   WHERE collection.archived = true
 )
 UPDATE collection

